hail/src/main/scala/is/hail/expr/ir/Emit.scala

```diff
@@ -969,6 +969,29 @@ private class Emit(
           sc.store,
           srvb.offset))
 
+      case CombOpValue(i, value, sig) =>
+        val AggContainer(_, sc) = container.get
+        val rvAgg = agg.Extract.getAgg(sig)
+        val newState = rvAgg.createState(mb.fb)
+
+        val t = value.pType.asInstanceOf[PBinary]
+        val xValue = emit(value)
+        EmitTriplet(
+          Code(xValue.setup,
+          xValue.m.mux(
+            Code._fatal("cannot combOp a missing value"),
+            Code(
+              newState.createState,
+              newState.deserializeFromBytes(t, coerce[Long](xValue.v)),
+              rvAgg.combOp(sc.states(i), newState)
```

Comment:
I don't remember how agg states work well enough. Does anything need to be done to free `newState`?

Reply:
It'll get cleaned up in RunAgg, I believe.